pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go

```diff
@@ -39,43 +39,69 @@ type EndpointBuilder struct {
 	tlsMode        string
 }
 
-func NewEndpointBuilder(c *Controller, pod *v1.Pod) *EndpointBuilder {
+func NewEndpointBuilder(c controllerInterface, pod *v1.Pod) *EndpointBuilder {
 	locality, sa, uid := "", "", ""
-	var podLabels labels.Instance
+	var endpointLabels labels.Instance
 	if pod != nil {
 		locality = c.getPodLocality(pod)
 		sa = kube.SecureNamingSAN(pod)
 		uid = createUID(pod.Name, pod.Namespace)
-		podLabels = pod.Labels
+		endpointLabels = pod.Labels
 	}
 
 	return &EndpointBuilder{
 		controller:     c,
-		labels:         podLabels,
+		labels:         augmentLabels(endpointLabels, c.Cluster(), locality),
```

Comment:
Why change it

Reply:
The motivation for adding labels is described in the RFC in the description.

Reply:
make sense we can support routing within  same topology